posthog/hogql/printer.py

```diff
@@ -1015,6 +1015,32 @@ def visit_call(self, node: ast.Call):
         if optimized_property_group_call := self.__get_optimized_property_group_call(node):
             return optimized_property_group_call
 
+        # Handle format strings in function names before checking function type
+        if func_meta := (
+            find_hogql_aggregation(node.name)
+            or find_hogql_function(node.name)
+            or find_hogql_posthog_function(node.name)
+        ):
+            if "{" in func_meta.clickhouse_name:
+                # Check if using positional arguments (e.g. {0}, {1})
+                if any(f"{{{i}}}" in func_meta.clickhouse_name for i in range(10)):  # Check common positions 0-9
+                    # if "splitByString" in func_meta.clickhouse_name:
+                    #     import ipdb; ipdb.set_trace()
+                    # For positional arguments, pass the args as a dictionary
+                    arg_arr = [self.visit(arg) for arg in node.args]
+                    try:
+                        return func_meta.clickhouse_name.format(*arg_arr)
+                    except (KeyError, IndexError) as e:
+                        raise QueryError(f"Invalid argument reference in function '{node.name}': {str(e)}")
+                else:
+                    # Original sequential placeholder behavior
+                    placeholder_count = func_meta.clickhouse_name.count("{}")
+                    if len(node.args) != placeholder_count:
+                        raise QueryError(
+                            f"Function '{node.name}' requires exactly {placeholder_count} argument{'s' if placeholder_count != 1 else ''} when using a format string"
```

Comment:
I'd suggest removing `when using a format string` from the error message - this means nothing to me as an end user

<img width="716" alt="image" src="https://github.com/user-attachments/assets/3032125e-6392-488d-a359-957ab09bd4dd" />